### PR TITLE
test: ErrorBoundary dedicated directory and tests

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR adds the tests for the ErrorBoundary component while moving it to its own dedicated directory.

### CODE REVIEW

**Comment by junlincc** [comment]
Thank you @geido for adding test to this component, which previously have no test coverage at all. 🙏 🙂

File: superset-frontend/src/components/ErrorBoundary/index.jsx [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { t } from '@superset-ui/core';
-import ErrorMessageWithStackTrace from './ErrorMessage/ErrorMessageWithStackTrace';
+import ErrorMessageWithStackTrace from '../ErrorMessage/ErrorMessageWithStackTrace';
```
Comment (nytai) [inline superset-frontend/src/components/ErrorBoundary/index.jsx:22]: nit: prefer absolute import unless it's `./`

Comment (nytai) [inline superset-frontend/src/components/ErrorBoundary/index.jsx:22]: nit: prefer absolute import unless it's `./`

**Review by nytai — APPROVED** [review]
lgtm, just minor nit. Let me know if you want to address or if I should merge as is.

**Review by rusackas — APPROVED** [review]

**Pull request merged**